packages/server/graphql/mutations/helpers/pushEstimateToGitLab.ts

```diff
@@ -0,0 +1,72 @@
+import {GraphQLResolveInfo} from 'graphql'
+import {SprintPokerDefaults} from 'parabol-client/types/constEnums'
+import makeAppURL from 'parabol-client/utils/makeAppURL'
+import appOrigin from '../../../appOrigin'
+import GitLabServerManager from '../../../integrations/gitlab/GitLabServerManager'
+import getPhase from '../../../utils/getPhase'
+import makeScoreGitLabComment from '../../../utils/makeScoreGitLabComment'
+import {GQLContext} from '../../graphql'
+import createNote from '../../nestedSchema/GitLab/mutations/createNote.graphql'
+import {ITaskEstimateInput} from '../../types/TaskEstimateInput'
+
+const pushEstimateToGitLab = async (
+  taskEstimate: ITaskEstimateInput,
+  context: GQLContext,
+  info: GraphQLResolveInfo,
+  stageId: string | undefined
+) => {
+  const {dimensionName, taskId, value, meetingId} = taskEstimate
+  const {dataLoader} = context
+  const [task, meeting] = await Promise.all([
+    dataLoader.get('tasks').load(taskId),
+    meetingId ? dataLoader.get('newMeetings').load(meetingId) : undefined
+  ])
+  const gitlabIntegration = task.integration as Extract<
+    typeof task.integration,
+    {service: 'gitlab'}
+  >
+  const {teamId} = task
+  const {accessUserId, gid} = gitlabIntegration
+  const [auth, fieldMap] = await Promise.all([
+    dataLoader
+      .get('teamMemberIntegrationAuths')
+      .load({service: 'gitlab', teamId, userId: accessUserId}),
+    dataLoader.get('gitlabDimensionFieldMaps').load({teamId, dimensionName, gid})
+  ])
+  if (!auth) return new Error('User no longer has access to GitLab')
+
+  const labelTemplate = fieldMap?.labelTemplate ?? SprintPokerDefaults.SERVICE_FIELD_COMMENT
+  if (labelTemplate === SprintPokerDefaults.SERVICE_FIELD_NULL) return undefined
+
+  const {accessToken, providerId} = auth
+  const provider = await dataLoader.get('integrationProviders').load(providerId)
+  if (!provider?.serverBaseUrl || !accessToken) return new Error('Invalid integration provider')
```

Comment:
Should we use `!` instead if it is not a possible scenario?

Reply:
As in `provider!.serverBaseUrl!`?

Reply:
@nickoferrall I mean consider serverBaseUrl non-nullable below and remove this check. Not sure if invalid integration provider is really possible
`const manager = new GitLabServerManager(accessToken, provider.serverBaseUrl!)`

Reply:
Also, above we checked `auth`: `if (!auth) return new Error('User no longer has access to GitLab')`
Could we have `accessToken` empty after that?

Reply:
maybe consider just:

```
if (!provider) return new Error('integration provider not found')
```

Reply:
@nickoferrall I meant something like

```
if (!auth) return new Error('User no longer has access to GitLab')
...
if (!provider) return new Error('integration provider not found')
...
const manager = new GitLabServerManager(accessToken!, provider.serverBaseUrl!)

```

But I think your approach is also fine

Reply:
Makes sense, thanks! Update pushed